environment/service_test.go

```diff
@@ -0,0 +1,137 @@
+package environment_test
+
+import (
+	"context"
+	"github.com/fabric8-services/fabric8-tenant/environment"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+	"testing"
+)
+
+var defaultLocationTempl = `apiVersion: v1
+kind: Template
+metadata:
+  name: fabric8-tenant-${DEPLOY_TYPE}
+objects:
+- apiVersion: v1
+  kind: ProjectRequest
+  metadata:
+    labels:
+      test: default-location
+      version: ${COMMIT}
+    name: ${USER_NAME}-${DEPLOY_TYPE}`
+
+var customLocationTempl = `apiVersion: v1
+kind: Template
+metadata:
+  name: fabric8-tenant-${DEPLOY_TYPE}
+objects:
+- apiVersion: v1
+  kind: ProjectRequest
+  metadata:
+    labels:
+      test: custom-location
+      version: ${COMMIT}
+    name: ${USER_NAME}-${DEPLOY_TYPE}`
+
+func TestGetAllTemplatesForAllTypes(t *testing.T) {
+	// given
+	service := environment.NewService(context.Background(), "", "", "")
+
+	for _, envType := range environment.DefaultEnvTypes {
+		// when
+		env, err := service.GetEnvData(envType)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, env.Name, envType)
+		if envType == "che" || envType == "jenkins" {
+			assert.Len(t, env.Templates, 2)
+			assert.Contains(t, env.Templates[0].Filename, envType)
+			assert.Contains(t, env.Templates[1].Filename, "quotas")
+		} else if envType == "user" {
+			assert.Len(t, env.Templates, 1)
+			assert.Contains(t, env.Templates[0].Filename, envType)
+		} else {
+			assert.Len(t, env.Templates, 1)
+			assert.Contains(t, env.Templates[0].Filename, "deploy")
+		}
+
+		for _, template := range env.Templates {
+			assert.NotEmpty(t, template.Content)
+		}
+	}
+}
+
+func TestDownloadFromGivenBlob(t *testing.T) {
+	// given
+	defer gock.OffAll()
+	gock.New("https://github.com").
+		Get("fabric8-services/fabric8-tenant/blob/123abc/environment/templates/fabric8-tenant-deploy.yml").
+		Reply(200).
+		BodyString(defaultLocationTempl)
+
+	service := environment.NewService(context.Background(), "", "123abc", "")
+
+	// when
+	envData, err := service.GetEnvData("run")
+
+	// then
+	require.NoError(t, err)
```

Comment:
can we add assert to check template.content string has some part from `defaultLocationTempl`

Reply:
I think that this should be enough: https://github.com/fabric8-services/fabric8-tenant/pull/634/files/ba57cf813296621b2405d470509c7f97833ee3e9#diff-1f307f420e41ce4eee63dc682f3640f4R89